00_download_data/1B_preprocess_methylation_data.ipynb

```diff
@@ -0,0 +1,444 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "(11985, 11882)\n"
+     ]
+    },
+    {
+     "data": {
+      "text/html": [
+       "<div>\n",
+       "<style scoped>\n",
+       "    .dataframe tbody tr th:only-of-type {\n",
+       "        vertical-align: middle;\n",
+       "    }\n",
+       "\n",
+       "    .dataframe tbody tr th {\n",
+       "        vertical-align: top;\n",
+       "    }\n",
+       "\n",
+       "    .dataframe thead th {\n",
+       "        text-align: right;\n",
+       "    }\n",
+       "</style>\n",
+       "<table border=\"1\" class=\"dataframe\">\n",
+       "  <thead>\n",
+       "    <tr style=\"text-align: right;\">\n",
+       "      <th>Composite Element REF</th>\n",
+       "      <th>cg00005847</th>\n",
+       "      <th>cg00008493</th>\n",
+       "      <th>cg00012199</th>\n",
+       "      <th>cg00012386</th>\n",
+       "      <th>cg00012792</th>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>sample_id</th>\n",
+       "      <th></th>\n",
+       "      <th></th>\n",
+       "      <th></th>\n",
+       "      <th></th>\n",
+       "      <th></th>\n",
+       "    </tr>\n",
+       "  </thead>\n",
+       "  <tbody>\n",
+       "    <tr>\n",
+       "      <th>TCGA-01-0628-11</th>\n",
+       "      <td>0.568261</td>\n",
+       "      <td>0.953532</td>\n",
+       "      <td>0.035959</td>\n",
+       "      <td>0.021465</td>\n",
+       "      <td>0.024005</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>TCGA-01-0630-11</th>\n",
+       "      <td>0.620646</td>\n",
+       "      <td>0.959144</td>\n",
+       "      <td>0.037645</td>\n",
+       "      <td>0.016915</td>\n",
+       "      <td>0.028272</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>TCGA-01-0631-11</th>\n",
+       "      <td>0.430550</td>\n",
+       "      <td>0.963075</td>\n",
+       "      <td>0.036071</td>\n",
+       "      <td>0.017380</td>\n",
+       "      <td>0.023571</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>TCGA-01-0633-11</th>\n",
+       "      <td>0.416541</td>\n",
+       "      <td>0.959544</td>\n",
+       "      <td>0.038710</td>\n",
+       "      <td>0.025398</td>\n",
+       "      <td>0.021352</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>TCGA-01-0636-11</th>\n",
+       "      <td>0.695484</td>\n",
+       "      <td>0.961997</td>\n",
+       "      <td>0.038338</td>\n",
+       "      <td>0.019061</td>\n",
+       "      <td>0.017349</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "Composite Element REF  cg00005847  cg00008493  cg00012199  cg00012386  \\\n",
+       "sample_id                                                               \n",
```

Comment:
I'm confused by what's going on here. Why is the sample ID on its own line? Shouldn't it just be a header column like the rest?

Reply:
I think that's just how the notebook formats `df.columns.name` (i.e. the columns name, or in this case it was the index name until I transposed the dataframe). The columns name doesn't get written as part of `df.to_csv`; the index name will just be a header column there like you're suggesting.

I'm not worried about this - I know the columns are methylation CpG probe IDs, and labeling them with the TCGA default `Composite Element REF` doesn't really tell me much.